### PR TITLE
fix for issue #3

### PR DESCRIPTION
**Pull request by sanjusoftware**
@tomzo I had to change the yamlbeans dependency repo location as I could not get authorization to the original maven repo to release a new jar there. I created my own repository com.github.sanjusoftware.yamlbeans and release the new 1.11 version of yaml beans there. used the same as dependency here in this release instead. Hope there is no concerns for this change.

this should fix issue #3

### CODE REVIEW

File: src/main/java/cd/go/plugin/config/yaml/YamlFileParser.java [hunk 1]
```diff
@@ -22,10 +23,14 @@ public JsonConfigCollection parseFiles(File baseDir, String[] files) {
...
                 inputStream = new FileInputStream(new File(baseDir, file));
-                YamlReader reader = new YamlReader(new InputStreamReader(inputStream));
+                YamlConfig config = new YamlConfig();
+                config.setAllowDuplicates(false);
+                YamlReader reader = new YamlReader(new InputStreamReader(inputStream), config);
                 Object rootObject = reader.read();
                 JsonConfigCollection filePart = rootTransform.transform(rootObject, file);
                 collection.append(filePart);
+            } catch (YamlReader.YamlReaderException ex) {
+                collection.addError("YAML contains duplicate keys: " + ex.getMessage() , file);
```
Comment (tomzo) [inline src/main/java/cd/go/plugin/config/yaml/YamlFileParser.java]: I presume that `YamlReader.YamlReaderException` is thrown also in other cases, when there is anything else wrong with yaml format, isn't it?
That catch would wrongly report such case as duplicate key, misleading user to think that there is some duplicate, while actually something else failed in parser.
Can you somehow identify that exception cause was actually a duplicate? Even checking the message in `ex` to contain `"duplicate"` would be fine.
Ideally there should be another test case with invalid yaml, which is not a duplicate, asserting sensible error message.

**Comment by tomzo** [comment]
>  I had to change the yamlbeans repo information as I could not get access to the original maven repo to release a new jar there

OK, I will have a look at that. I'd prefer to stay with official, but if it is not there, I will use yours on next release.

I added a small suggestion in code. 
Everything looks great though, thanks for your work :+1:

File: src/main/java/cd/go/plugin/config/yaml/YamlFileParser.java (hunk 1 above)
Comment (sanjusoftware) [inline src/main/java/cd/go/plugin/config/yaml/YamlFileParser.java]: you are right!! how about just directly adding the YamlReaderException (or better YamlException) message, whatever it is and surface it to the user. This would also help users fix if there are other genuine issues with their YAML caught by YamlBeans? e.g below:

```java
} catch (YamlReader.YamlReaderException ex) {
   collection.addError(ex.getMessage() , file);
}
````

The assertion in test that I added in that case becomes:

````java
assertThat(errors.get(0).getAsJsonObject().getAsJsonPrimitive("message").getAsString(), is("Line 9, column 20: Duplicate key found 'upstream'"));
// "Line 9, column 20: Duplicate key found 'upstream'" is directly thrown by YamlBeans library now
````

**Comment by sanjusoftware** [comment]
> OK, I will have a look at that. I'd prefer to stay with official, but if it is not there, I will use yours on next release.

makes sense. BTW, if you look at it, there is nothing unofficial about it anyways except that earlier a contributor called Tony release the jar on mvn and now I am releasing. But you are right, the maven central groupId is changing, which is best if it can remain same as older.

FYI: [here](https://oss.sonatype.org/#nexus-search;quick~yamlbeans) and [here] (http://search.maven.org/#search%7Cga%7C1%7Cyamlbeans) are the new mvn jar locations

File: src/main/java/cd/go/plugin/config/yaml/YamlFileParser.java (hunk 1 above)
Comment (sanjusoftware) [inline src/main/java/cd/go/plugin/config/yaml/YamlFileParser.java]: ok.. updating then!! 👍
